src/stores/appcatalog/actions.ts

```diff
@@ -23,3 +29,66 @@ export const listCatalogs = createAsynchronousAction<any, any, any>({
   shouldPerform: () => true,
   throwOnError: false,
 });
+
+export const installLatestIngress = createAsynchronousAction<
+  IInstallIngress,
+  IState,
+  void
+>({
+  actionTypePrefix: 'INSTALL_INGRESS_APP',
+  perform: async (state, dispatch, payload) => {
+    if (payload?.clusterId) {
+      const gsCatalog = selectIngressCatalog(state);
+
+      const { name, version } = gsCatalog.apps[
+        Constants.INSTALL_INGRESS_TAB_APP_NAME
+      ][0];
+
+      const appToInstall = {
+        name,
+        chartName: name,
+        version,
+        catalog: Constants.INSTALL_INGRESS_TAB_APP_CATALOG_NAME,
+        namespace: 'kube-system',
+        valuesYAML: '',
+        secretsYAML: '',
+      };
+
+      /* eslint-disable @typescript-eslint/await-thenable */
+      await dispatch(
+        installApp({ app: appToInstall, clusterId: payload.clusterId })
+      );
+      await dispatch(loadClusterApps({ clusterId: payload.clusterId }));
+      /* eslint-enable @typescript-eslint/await-thenable */
+    }
+  },
+  shouldPerform: (state) => {
+    // only allow performing if we have loaded the catalog and have a version
+    const gsCatalog = selectIngressCatalog(state);
+
+    const app = gsCatalog?.apps?.[Constants.INSTALL_INGRESS_TAB_APP_NAME]?.[0];
+
+    return gsCatalog && app;
+  },
+  throwOnError: false,
+});
+
+export const prepareIngressTabData = createAsynchronousAction<
+  IInstallIngress,
+  IState,
+  void
+>({
+  actionTypePrefix: 'PREPARE_INGRESS_TAB_DATA',
+  perform: async (state, dispatch, payload) => {
+    if (payload?.clusterId) {
+      const gsCatalog = selectIngressCatalog(state);
+
+      await Promise.all([
+        dispatch(loadClusterApps({ clusterId: payload.clusterId })),
+        catalogLoadIndex(gsCatalog)(dispatch, () => state),
```

Comment:
```suggestion
        dispatch(catalogLoadIndex(gsCatalog)),
```

Reply:
It seems you commented on an outdated revision. The Github UI tells me `Outdated suggestions cannot be applied`

Reply:
GitHub strikes again 😅